content/200-concepts/100-components/01-prisma-schema/10-postgresql-extensions.mdx

```diff
@@ -0,0 +1,108 @@
+
+To install an extension, it must already be available on the local file system of your database server. Installing the extension then runs a script file that adds the new functionality.
+
+Prisma's `postgresqlExtensions` preview feature allows you to represent PostgreSQL extensions in your Prisma schema. Note that specific extensions may add functionality that is not currently supported by Prisma. For example, an extension may add a type or index that is not supported by Prisma. This functionality must be implemented on a case-by-case basis and is not provided by this preview feature.
+
+## How to enable the <inlinecode>postgresqlExtensions</inlinecode> preview feature
+
+Representing PostgreSQL extensions in your Prisma schema is currently a preview feature. To enable the `postgresqlExtensions` preview feature, you will need to add the `postgresqlExtensions` feature flag to the `previewFeatures` field of the `generator` block in your Prisma schema file:
+
+```prisma file=schema.prisma highlight=3;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["postgresqlExtensions"]
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+```
+
+## How to represent PostgreSQL extensions in your Prisma schema
+
+To represent PostgreSQL extensions in your Prisma schema, add the `extensions` field to the `datasource` block of your `schema.prisma` file with an array of the extensions that you require. For example, the following schema lists the `hstore`, `pg_tgrm` and `postgis` extensions:
+
+```prisma file=schema.prisma
+datasource db {
+  provider = "postgresql"
+  url      = env("TEST_DATABASE_URL")
+  extensions = [hstore(schema: "myHstoreSchema"), pg_tgrm, postgis(version: "2.1")]
+}
+```
+
+Each extension name in the Prisma schema can take the following optional arguments:
+
+- `schema`: the name of the schema in which to install the extension's objects. If this argument is not specified, the current default object creation schema is used.
+- `version`: the version of the extension to install. If this argument is not specified, the value given in the extension's control file is used.
+- `map`: the database name of the extension. If this argument is not specified, the name of the extension in the Prisma schema must match the database name.
+
+In the example above, the `hstore` extension uses the `myHstoreSchema` schema, and the `postgis` extension is installed with version 2.1 of the extension.
+
+The `map` argument is useful when the PostgreSQL extension that you want to install has a name that is not a valid identifier in the Prisma schema. For example, the `uuid-ossp` PostgreSQL extension name is an invalid identifier because it contains a hyphen. In the following example, the extension is mapped to the valid name `uuidOssp` in the Prisma schema:
```

Comment:
```suggestion
The `map` argument is useful when the PostgreSQL extension that you want to activate has a name that is not a valid identifier in the Prisma schema. For example, the `uuid-ossp` PostgreSQL extension name is an invalid identifier because it contains a hyphen. In the following example, the extension is mapped to the valid name `uuidOssp` in the Prisma schema:
```